src/bigint.rs

```diff
@@ -1891,7 +1901,15 @@ impl Num for BigInt {
     /// Creates and initializes a BigInt.
     #[inline]
     fn from_str_radix(mut s: &str, radix: u32) -> Result<BigInt, ParseBigIntError> {
-        let sign = if s.starts_with('-') { s = &s[1..]; Minus } else { Plus };
+        let sign = if s.starts_with('-') {
+            s = &s[1..];
+            if s.starts_with('+') {
+                // create ParseIntError::InvalidDigit
+                let e = i64::from_str_radix("-+1", radix).unwrap_err();
+                return Err(e.into());
+            }
```

Comment:
What if instead, you only take the substring consuming '-' when it's not followed by '+'.  So when there is a "-+", it just passes the whole thing to BigUint, which will barf on '-'.

    if s.starts_with('-') {
        // Only consume the minus sign if it's alone
        let tail = &s[1..];
        if !tail.starts_with('+') { s = tail; }
    ...